src/growable.c

```diff
@@ -0,0 +1,276 @@
+  R_set_altvec_Dataptr_method(dta_grow_integer, altinteger_Dataptr);
+  R_set_altvec_Dataptr_or_null_method(dta_grow_integer, altall_Dataptr_or_null);
+  R_set_altvec_Extract_subset_method(dta_grow_integer, altinteger_Extract_subset);
+  R_set_altinteger_Get_region_method(dta_grow_integer, altinteger_Get_region);
+  dta_grow_logical = R_make_altlogical_class("growable_logical_v0", "data.table", info);
+  R_set_altrep_Length_method(dta_grow_logical, altall_Length);
+  R_set_altrep_Inspect_method(dta_grow_logical, altlogical_Inspect);
+  R_set_altvec_Dataptr_method(dta_grow_logical, altlogical_Dataptr);
+  R_set_altvec_Dataptr_or_null_method(dta_grow_logical, altall_Dataptr_or_null);
+  R_set_altvec_Extract_subset_method(dta_grow_logical, altlogical_Extract_subset);
+  R_set_altlogical_Get_region_method(dta_grow_logical, altlogical_Get_region);
+  dta_grow_real = R_make_altreal_class("growable_real_v0", "data.table", info);
+  R_set_altrep_Length_method(dta_grow_real, altall_Length);
+  R_set_altrep_Inspect_method(dta_grow_real, altreal_Inspect);
+  R_set_altvec_Dataptr_method(dta_grow_real, altreal_Dataptr);
+  R_set_altvec_Dataptr_or_null_method(dta_grow_real, altall_Dataptr_or_null);
+  R_set_altvec_Extract_subset_method(dta_grow_real, altreal_Extract_subset);
+  R_set_altreal_Get_region_method(dta_grow_real, altreal_Get_region);
+  dta_grow_complex = R_make_altcomplex_class("growable_complex_v0", "data.table", info);
+  R_set_altrep_Length_method(dta_grow_complex, altall_Length);
+  R_set_altrep_Inspect_method(dta_grow_complex, altcomplex_Inspect);
+  R_set_altvec_Dataptr_method(dta_grow_complex, altcomplex_Dataptr);
+  R_set_altvec_Dataptr_or_null_method(dta_grow_complex, altall_Dataptr_or_null);
+  R_set_altvec_Extract_subset_method(dta_grow_complex, altcomplex_Extract_subset);
+  R_set_altcomplex_Get_region_method(dta_grow_complex, altcomplex_Get_region);
+  dta_grow_raw = R_make_altraw_class("growable_raw_v0", "data.table", info);
+  R_set_altrep_Length_method(dta_grow_raw, altall_Length);
+  R_set_altrep_Inspect_method(dta_grow_raw, altraw_Inspect);
+  R_set_altvec_Dataptr_method(dta_grow_raw, altraw_Dataptr);
+  R_set_altvec_Dataptr_or_null_method(dta_grow_raw, altall_Dataptr_or_null);
+  R_set_altvec_Extract_subset_method(dta_grow_raw, altraw_Extract_subset);
+  R_set_altraw_Get_region_method(dta_grow_raw, altraw_Get_region);
+  dta_grow_list = R_make_altlist_class("growable_list_v0", "data.table", info);
+  R_set_altrep_Length_method(dta_grow_list, altall_Length);
+  R_set_altrep_Inspect_method(dta_grow_list, altlist_Inspect);
+  R_set_altvec_Dataptr_method(dta_grow_list, altlist_Dataptr);
+  R_set_altvec_Dataptr_or_null_method(dta_grow_list, altall_Dataptr_or_null);
+  R_set_altlist_Elt_method(dta_grow_list, altlist_Elt);
+  R_set_altlist_Set_elt_method(dta_grow_list, altlist_Set_elt);
+}
+
+static R_altrep_class_t dta_grow_string, dta_grow_integer, dta_grow_logical, dta_grow_real, dta_grow_complex, dta_grow_raw, dta_grow_list;
```

Comment:
We already have the same line in line 48?

Reply:
Good catch, thanks!